src/public/modules/forms/helpers/__tests__/CsvGenerator.spec.ts

```diff
@@ -0,0 +1,192 @@
+import { stringify } from 'csv-string'
+import { mocked } from 'ts-jest/utils'
+
+import { CsvGenerator } from '../CsvGenerator'
+import { triggerFileDownload } from '../util'
+
+const UTF8_BYTE_ORDER_MARK = '\uFEFF'
+
+jest.mock('../util')
+const MockTriggerFileDownload = mocked(triggerFileDownload)
+
+describe('CsvGenerator', () => {
+  afterAll(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('Constructor', () => {
+    it('should initialise the instance properties correctly with defaults if params are not passed to constructor', () => {
+      // Arrange
+
+      const csv = new CsvGenerator()
+
+      // Assert
+
+      expect(csv.expectedNumberOfRecords).toEqual(0)
+      expect(csv.numOfMetaDataRows).toEqual(0)
+      expect(csv.numOfHeaderRows).toEqual(1)
+      expect(csv.lastCreatedAt).toEqual(0)
+      expect(csv.startIdx).toEqual(2)
+      expect(csv.idx).toEqual(2)
+      expect(csv.records.length).toEqual(2)
+      expect(csv.records[0]).toEqual(UTF8_BYTE_ORDER_MARK)
+    })
+
+    it('should initialise the instance properties correctly if params are passed to constructor', () => {
+      // Arrange
+
+      const csv = new CsvGenerator(1, 2)
```

Comment:
should we assign the numbers in the constructor (`1`, `2`) to a constant so it is more obvious what we are looking at?

e.g.

```ts
const expectedNumberOfRecords = 1, numOfMetaDataRows = 2

...

expect(csv.expectedNumberOfRecords).toEqual(expectedNumberOfRecords)
expect(csv.numOfMetaDataRows).toEqual(numOfMetaDataRows)

// More obvious calculations for the other props too
expect(csv.startIdx).toEqual(expectedNumberOfRecords + numOfMetaDataRows + csv.numOfHeaderRows)
...

```

Reply:
thanks, edited